dll/win32/uxtheme/nonclient.c

```diff
@@ -1075,6 +1078,118 @@ ThemeWndProc(HWND hWnd, UINT Msg, WPARAM wParam, LPARAM lParam, WNDPROC DefWndPr
+
+    RECT rcWindowPrev = { pcontext->wi.rcWindow.left, pcontext->wi.rcWindow.top, pcontext->wi.rcWindow.right, pcontext->wi.rcWindow.bottom };
+    RECT rcClientPrev = { pcontext->wi.rcClient.left, pcontext->wi.rcClient.top, pcontext->wi.rcClient.right, pcontext->wi.rcClient.bottom };
+    SetWindowPos(pcontext->hWnd, NULL, left, top, right - left, bottom - top, SWP_NOZORDER | SWP_NOACTIVATE | SWP_DRAWFRAME | SWP_NOCOPYBITS);
+    RECT rcWindowNew = { left, top, right, bottom };
+    pcontext->wi.rcWindow = rcWindowNew;
+
+    BOOL hasVScrollBar = dwStyle & WS_VSCROLL;
+    if (hasVScrollBar)
+    {
+        EnableScrollBar(pcontext->hWnd, SB_VERT, ESB_ENABLE_BOTH);
+
+        dummyScrollInfo.cbSize = sizeof(SCROLLINFO);
+        dummyScrollInfo.fMask = SIF_DISABLENOSCROLL | SIF_POS | SIF_RANGE;
+        dummyScrollInfo.nMin = 0;
+        dummyScrollInfo.nMax = rcWindowNew.bottom - rcWindowNew.top;
+        dummyScrollInfo.nPos = 0;
+        SetScrollInfo(pcontext->hWnd, SB_VERT, &dummyScrollInfo, TRUE);
+    }
+
+    SetViewportOrgEx(hDC, rcWindowNew.left, rcWindowNew.top, NULL);
+
+    INT offsetX = -rcWindowNew.left;
+    INT offsetY = -rcWindowNew.top;
+    OffsetRect(&rcWindowNew, offsetX, offsetY);
+    ThemeCalculateCaptionButtonsPosEx(pcontext->wi, pcontext->hWnd, pcontext->theme, pcontext->CaptionHeight - pcontext->wi.cyWindowBorders);
+
+
+    INT leftBorderInset = pcontext->wi.cxWindowBorders;
+    INT titleBarInset = pcontext->CaptionHeight; // + pcontext->wi.cyWindowBorders;
+    INT rightBorderInset = pcontext->wi.cxWindowBorders;
+    INT bottomBorderInset = pcontext->wi.cyWindowBorders;
+
+    RECT rcClientNew;
+    if (GetWindowRect(pcontext->hWnd, &rcClientNew))
+    {
+        rcClientNew.left += leftBorderInset;
+        rcClientNew.top += titleBarInset;
+        rcClientNew.right -= rightBorderInset;
+        rcClientNew.bottom -= bottomBorderInset;
+    }
+    pcontext->wi.rcClient = rcClientNew;
+
+    pcontext->wi.dwStyle &= ~(WS_HSCROLL | WS_VSCROLL);
+    ThemePaintWindow(pcontext, &rcWindowNew, FALSE);
+    pcontext->wi.dwStyle = dwStyle;
+
+    if (hasVScrollBar && IsScrollBarVisible(pcontext->hWnd, OBJID_VSCROLL))
+    {
+        SCROLLBARINFO sbi;
+        sbi.cbSize = sizeof(sbi);
+        GetScrollBarInfo(pcontext->hWnd, OBJID_VSCROLL, &sbi);
+        INT scWidth = sbi.rcScrollBar.right - sbi.rcScrollBar.left;
+
+        sbi.rcScrollBar.right = rcClientNew.right;
+        rcClientNew.right -= scWidth;
+        sbi.rcScrollBar.left = rcClientNew.right;
+
+        sbi.rcScrollBar.top = rcClientNew.top;
+        sbi.rcScrollBar.bottom = rcClientNew.bottom;
+
+        ThemeDrawScrollBarEx(pcontext, SB_VERT, &sbi, NULL);
+    }
+    pcontext->wi.rcClient = rcClientNew;
+
+    OffsetRect(&rcClientNew, -pcontext->wi.rcWindow.left, -pcontext->wi.rcWindow.top);
+    
+    if (drawClientAreaColor)
+        FillRect(hDC, &rcClientNew, (HBRUSH)(COLOR_WINDOW + 1));
```

Comment:
Ahh. Now I see why maybe the Rectangle function was originally used :oof: because one should use the theme-specific `COLOR_WINDOW` color, and not the currently active one.
However i think you still can use the FillRect function as follows:
```suggestion
    if (drawClientAreaColor)
    {
        HBRUSH hbrWindow = GetThemeSysColorBrush(pcontext->theme, COLOR_WINDOW);
        FillRect(hDC, &rcClientNew, hbrWindow);
        DeleteObject(hbrWindow);
    }
```